libraries/app/database_api.cpp

```diff
@@ -362,6 +363,24 @@ optional<block_header> database_api_impl::get_block_header(uint32_t block_num) c
       return *result;
    return {};
 }
+map<uint32_t,block_header> database_api::get_block_header_batch(const vector<uint32_t> block_num)const
+{
+   return my->get_block_header_batch( block_num );
+}
+
+map<uint32_t,block_header> database_api_impl::get_block_header_batch(const vector<uint32_t> block_num) const
+{
+   map<uint32_t,block_header> results;
+   for (const uint32_t& i : block_num) {
+      auto result = _db.fetch_block_by_number(i);
+      if (result)
+         results[i] = *result;
```

Comment:
Please excuse this comment - I am not much a C++ dev - but shouldn't we prefer `results.push_back(result)` over using pointers?